examples/context-propagation/kotlin/src/main/kotlin/example/armeria/contextpropagation/kotlin/MainService.kt

```diff
@@ -0,0 +1,126 @@
+        val ctxExecutor = ctx.contextAwareExecutor()
+        val response = GlobalScope.future(ctxExecutor.asCoroutineDispatcher()) {
+
+            val numsFromRequest = async { fetchFromRequest(ctx, req) }
+            val numsFromDb = async { fetchFromFakeDb(ctx) }
+            val nums = awaitAll(numsFromRequest, numsFromDb).flatten()
+
+            // The context is kept after resume.
+            require(ServiceRequestContext.current() === ctx)
+            require(ctx.eventLoop().inEventLoop())
+
+            val backendResponses =
+                awaitAll(
+                    *nums.map { num ->
+                        // The context is mounted in a thread-local, meaning it is available to all logic such
+                        // as tracing.
+                        require(ServiceRequestContext.current() === ctx)
+                        require(ctx.eventLoop().inEventLoop())
+
+                        backendClient.get("/square/$num").aggregate().asDeferred()
+                    }.toTypedArray()
+                ).toList()
+
+            // The context is mounted in a thread-local, meaning it is available to all logic such as tracing.
+            require(ServiceRequestContext.current() === ctx)
+            require(ctx.eventLoop().inEventLoop())
+
+            HttpResponse.of(
+                backendResponses.stream()
+                    .map(AggregatedHttpResponse::contentUtf8)
+                    .collect(Collectors.joining("\n"))
+            )
+        }
+        return HttpResponse.from(response)
+    }
+
+    private suspend fun fetchFromRequest(ctx: ServiceRequestContext, req: HttpRequest): List<Long> {
+        // The context is mounted in a thread-local, meaning it is available to all logic such as tracing.
+        require(ServiceRequestContext.current() === ctx)
+        require(ctx.eventLoop().inEventLoop())
+
+        val aggregatedHttpRequest = req.aggregate().await()
+
+        // The context is kept after resume.
+        require(ServiceRequestContext.current() === ctx)
+        require(ctx.eventLoop().inEventLoop())
+
+        val nums = mutableListOf<Long>()
+        for (token in Iterables.concat(
+            NUM_SPLITTER.split(aggregatedHttpRequest.path().substring(1)),
+            NUM_SPLITTER.split(aggregatedHttpRequest.contentUtf8())
+        )) {
+            nums.add(token.toLong())
+        }
+        return nums
+    }
+
+    private suspend fun fetchFromFakeDb(ctx: ServiceRequestContext): List<Long> {
+        // The context is mounted in a thread-local, meaning it is available to all logic such as tracing.
+        require(ServiceRequestContext.current() === ctx)
+        require(ctx.eventLoop().inEventLoop())
+
+        // This logic mimics using a blocking method, which would usually be something like a MySQL
+        // database query using JDBC.
+        return CompletableFuture.supplyAsync(
+            Supplier<List<Long>> {
```

Comment:
Maybe this type came automatically from intellij

Reply:
Looks like kotlin failed to infer the type, so I need to add it explicitly.